docs/source/external_exporters.rst

```diff
@@ -0,0 +1,156 @@
+
+Under the hood exporters are python classes that expose a certain interface. 
+Any importable classes that expose this interface can be use as an exporter for
+nbconvert. 
+
+For simplicity we expose basic classes that implement all the relevant methods
+that you have to subclass and overwrite just the relevant methods to provide a
+custom exporter. Below we show you the step to create a custom exporter that
+provides a custom file extension, and a custom template that inserts before and after
+each markdown cell.
+
+We will lay out files to be ready for Python packaging and distributing on PyPI, 
+although the exact art of Python packaging is beyond the scope of this explanation. 
+
+We will use the following layout for our package to expose a custom exporter::
+
+    mypackage
+    ├── LICENSE.md
+    ├── setup.py
+    └── mypackage
+        ├── __init__.py
+        └── templates
+            └── test_template.tpl
+
+As you can see the layout is relatively simple, in the case where a template is not 
+needed we would actually have only one file with an Exporter implementation.  Of course 
+you can change the layout of your package to have a more fine-grained structure of the 
+subpackage. But lets see what a minimum example looks like.
+
+We are going to write an exporter that:
+
+  - exports to html, so we will reuse the built-in html exporter
+  - changes the file extension to `.test_ext`
+
+::
+    # file __init__.py
+    import os
+    import os.path
+
+    from traitlets.config import Config
+    from nbconvert.exporters.html import HTMLExporter
+
+    #-----------------------------------------------------------------------------
+    # Classes
+    #-----------------------------------------------------------------------------
+
+    class MyExporter(HTMLExporter):
+        """
+        My custom exporter  
+        """
+        
+        def _file_extension_default(self):
+            """
+            The new file extension is `.test_ext`
+            """
+            return '.test_ext'
+
+        @property
+        def template_path(self):
+            """
+            We want to inherit from HTML template, and have template under
+            `./templates/` so append it to the search path. (see next section)
+            """
+            return super().template_path+[os.path.join(os.path.dirname(__file__), "templates")]
+
+        def _template_file_default(self):
+            """
+            We want to use the new template we ship with our library.
+            """
+            return 'test_template' # full
+        
+
+And the template file, that inherits from the html `full` template and prepend/append text to each markdown cell (see Jinja2 docs for template syntax)::
+
+    {% extends "full.tpl" %}
+
+    {% block markdowncell -%}
+
+
+    ## this is a markdown cells
+    {super()}
+    ## THIS IS THE END
+
+
+    {% endblock markdowncell %}
+
+
+Assuming you install this pacakge locally, or from PyPI, you can now use::
```

Comment:
typo package

@Carreau This is a really helpful doc. I spent the morning reading the nbconvert source. Love this addition.